validate healthcheck params in daemon side

allencloud opened the pull request:
Signed-off-by: allencloud <allen.sun@daocloud.io>

fixes #30202 

validate user input healthcheck parameters via API in the daemon side. 

**- What I did**
1. validate healthcheck in daemon side
2. add a test for api testing invalid healthcheck parameters
3. add missing `--health-retries` validation in client side

**- How I did it**

**- How to verify it**

**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->


**- A picture of a cute animal (not mandatory but encouraged)**

allencloud committed:
validate healthcheck params in daemon side

Signed-off-by: allencloud <allen.sun@daocloud.io>

cpuguy83 commented on daemon/container.go:
```diff
@@ -237,6 +237,21 @@ func (daemon *Daemon) verifyContainerSettings(hostConfig *containertypes.HostCon
+
+		// Validate the healthcheck params of Config
+		if config.Healthcheck != nil {
+			if config.Healthcheck.Interval < 0 {
```
I think this should validate less than `1s`, wdyt as I don't think we can guarantee sub-second checking.

allencloud replied:
If that, maybe we need to use a consistent validation in the whole procedure. I think there are three places to validate:
1. client side;
2. daemon received parameters from API;
3. parameters from dockerfile when building

I think consistency is the first thing(no matter less than 0 or less than 1s), the second thing to add more specific explanation in docs. 

But currently, I am on less than 0.  😄 Open for input.

cpuguy83 replied:
Sure, consistent is good, but the daemon side is the "important" piece.
I just think if we are going to add a validation, let's make sure it's not letting non-sensical values.

LK4D4 commented:
@cpuguy83 there is question for you here :)

allencloud commented:
Hi @cpuguy83 @LK4D4 
Thanks for the review. Currently I updated this PR with modifying the validation from `0` to `1s`. And in additional, I added more specific details in docs swagger.yml and update api 1.24, since healthcheck is introduced in docker 1.12.x and api 1.24. PTAL, thanks 😄

LK4D4 commented:
LGTM

cpuguy83 reviewed (approved):
LGTM
